drivers/zfs/zfs.go

```diff
@@ -392,12 +392,19 @@ func (d *Driver) Remove(id string) error {
 	name := d.zfsPath(id)
 	dataset := zfs.Dataset{Name: name}
 	err := dataset.Destroy(zfs.DestroyRecursive)
-	if err == nil {
-		d.Lock()
-		delete(d.filesystemsCache, name)
-		d.Unlock()
+	if err != nil {
+		// We must be tolerant in case the image has already been removed,
+		// for example, accidentally by hand.
+		mountpoint := d.mountPath(id)
```

Comment:
I think I should have used `zfsPath` instead of `mountPath`, since `mountPath` is intended for the mount point of a ZFS dataset.